### ISSUE TITLE
CrossValidation RegressionMetrics has two more entries than number of folds

### ISSUE THREAD

**Issue by jwood803** [dotnet/machinelearning#570]
I might be incorrect on this, but wouldn't the number of `RegressionMetrics` be the same as the number of folds specified in the `CrossValidator`?

### System information
- Windows 10
- **.NET Version (eg., dotnet --info)**: 
```
.NET Command Line Tools (2.1.202)

Product Information:
 Version:            2.1.202
 Commit SHA-1 hash:  281caedada

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.202\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.9
  Build    : 1632fa1589b0eee3277a8841ce1770e554ece037
```
### Issue

- **What did you do?**
Ran cross validation against a pipeline.
- **What happened?**
The `RegressionMetrics` count is two more than the number of folds specified.
- **What did you expect?**
The `RegressionMetrics` count to be the same as the number of folds.

### Source code / logs
```csharp
var pipeline = new LearningPipeline
{
    new TextLoader(dataset).CreateFrom<SalaryData>(useHeader: true, separator: ','),
    new ColumnConcatenator("Features", "YearsExperience"),
    new GeneralizedAdditiveModelRegressor()
};

var crossValidator = new CrossValidator()
{
    Kind = MacroUtilsTrainerKinds.SignatureRegressorTrainer,
    NumFolds = 2
};
var crossValidatorOutput = crossValidator.CrossValidate<SalaryData, SalaryPrediction>(pipeline);

crossValidatorOutput.RegressionMetrics.ForEach(m => Console.WriteLine(m.Rms));
```

![2018-07-20 14_44_23-clipboard](https://user-images.githubusercontent.com/1578160/43036237-6c8bf9a8-8ccb-11e8-8a79-82f45ccd18cc.png)

![2018-07-20 14_45_12-clipboard](https://user-images.githubusercontent.com/1578160/43036238-73ff4eb0-8ccb-11e8-9f3b-e4c3b71cbbf9.png)

Looking at `sklearn`, it seems to have the same number of results as the number of folds:

```python
from sklearn.model_selection import cross_val_score

cross_val_score(lin_reg, train_set, train_labels, cv=3)
```

```bash
array([0.96044449, 0.97351702, 0.92777218])
```

**Comment by Zruty0** [comment]
I think this is by design. I might be wrong, but I suspect that the extra two metric rows could be the 'overall' metrics and their standard deviation? 

Maybe @yaeldekel can confirm or deny.

**Comment by codemzs** [comment]
@jwood803 @Zruty0 This is by design. The first two rows contain average and std. deviations of the metrics from all the folds.

**Comment by Ivanidzo4ka** [comment]
Not @yaeldekel , but can confirm.
I think one good solution is to expose "Fold Index" field which getting lost.

**Comment by Ivanidzo4ka** [comment]
@jwood803 Would you like to have these average and std deviations of metrics in returning result with a way to distinguish them from fold results, or you fine with just want fold results?
